x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -251,6 +253,29 @@ export class ActionExecutor {
 
         event.event = event.event || {};
 
+        // add event.kibana.action.execution.gen_ai to event log when GenerativeAi Connector is executed
+        if (result.status === 'ok' && actionTypeId === '.gen-ai') {
+          const data = result.data as unknown as { usage: {} };
+          event.kibana = event.kibana || {};
+          event.kibana.action = event.kibana.action || {};
+          event.kibana = {
+            ...event.kibana,
+            action: {
+              ...event.kibana.action,
+              execution: {
+                ...event.kibana.action.execution,
+                gen_ai: {
+                  usage: data.usage,
+                },
+              },
+            },
+          };
+        }
```

Comment:
could we add a blank line here?  To separate the gen_ai specific bits from the framework general bits.